module/VuFind/src/VuFind/Hierarchy/TreeRenderer/JSTree.php

```diff
@@ -220,10 +220,10 @@ protected function getContextualUrl($node, $context)
             if ('collection' === $type && !$this->collectionsEnabled) {
                 $type = 'record';
             }
-            $url = $this->getUrlFromRouteCache($type, $node->id);
+            $url = $this->getUrlFromRouteCache($type, urlencode($node->id));
```

Comment:
There are two calls to getUrlFromRouteCache, but you're only urlencoding one of them. Is that intentional? Would it make more sense to do the url-encoding inside the getUrlFromRouteCache function?

Reply:
Not intentional, I moved encoding to getUrlFromRouteCache.